v2/svid/jwtsvid/svid.go

```diff
@@ -57,6 +57,11 @@ func ParseAndValidate(token string, bundles jwtbundle.Source, audience []string)
 			return nil, jwtsvidErr.New("token header missing key id")
 		}
 
+		// forbid tokens which have the `typ` header, which is not either "JOSE" or "JWT"
+		if typ := tok.Headers[0].ExtraHeaders[jose.HeaderType]; typ != "JOSE" && typ != "JWT" {
```

Comment:
The spec states that this validation should only take place if the type claim is present:
```suggestion
		if typ, present := tok.Headers[0].ExtraHeaders[jose.HeaderType]; present && typ != "JOSE" && typ != "JWT" {
```